File: src/DSharp.Script/src/System/Threading/Tasks/TaskCompletionSource.js

function TaskCompletionSource_$1() {
    this._isCompleted = false;
    this.task = new promise((resolve, reject) => {
        this._reject = reject;
        this._resolve = resolve;
    });
}

var TaskCompletionSource_$1$ = {
    setResult: function (result) {
        if (!this._isCompleted) {
            this._isCompleted = true;
            this._resolve(result);
        }
    },
    setException: function (exception) {
        if (!this._isCompleted) {
            this._isCompleted = true;
            this._reject(exception);
        }
    },
    trySetResult: function (result) {
        if (!this._isCompleted) {
            this._isCompleted = true;
            this._resolve(result);
            return true;
        }
        return false;
    },
    trySetException: function (exception) {
        if (!this._isCompleted) {
            this._isCompleted = true;
            this._reject(exception);
            return true;
        }
        return false;
    },
};

File: src/DSharp.Mscorlib/Threading/Tasks/TaskCompletionSource.cs
namespace System.Threading.Tasks
{
    public sealed class TaskCompletionSource<TResult>
    {
        public Task<TResult> Task { get; }

        public void SetResult(TResult result)
        {
        }

        public void SetException(Exception ex)
        {
        }

        public bool TrySetResult(TResult result)
        {
            return true;
        }

        public bool TrySetException(Exception ex)
        {
            return true;
        }
    }
}
